table/tables/index.go

```diff
@@ -97,17 +97,15 @@ func (c *indexIter) Next() (val []types.Datum, h int64, err error) {
 
 // index is the data structure for index data in the KV store.
 type index struct {
-	tblInfo *model.TableInfo
 	idxInfo *model.IndexInfo
 	prefix  kv.Key
 }
 
 // NewIndex builds a new Index object.
-func NewIndex(tableInfo *model.TableInfo, indexInfo *model.IndexInfo) table.Index {
+func NewIndex(partitionID int64, indexInfo *model.IndexInfo) table.Index {
```

Comment:
I have a problem with the naming of `partitionID`. Is it always being `paritonID`? Is it possible being `table ID`?

Reply:
Agree. Please comment it.

Reply:
It may be partition ID or table  ID, depends on whether the table is a partitioned table or not.

Reply:
If that is the case, then `partitionID` is not a good choice. It would be nice if you could change `partitionID` to `ID` and add some comments like "ID could be partitionID or table ID".